### PR TITLE
Add migration documentation for next release

### PR DESCRIPTION
**Pull request by JulienPradet**
* [ ] Please make sure that the versions of the release and the Magento modules are correct before merging this branch.

### CODE REVIEW

File: source/docs/advanced/theme/analytics.md [hunk 1]
```diff
@@ -151,15 +151,27 @@ module.exports = {
...
+            "Google Analytics": {
+              trackingId: "UA-123-1",
+              anonymizeIp: !authorization,
+            },
+          };
         },
+
         // integration that will add itself to the `analytics.js` lib
         script: () =>
           require("@segment/analytics.js-integration-google-analytics")
```
Comment (JulienPradet) [inline source/docs/advanced/theme/analytics.md:177]: We should use an import syntax here.

Comment (real34) [reply source/docs/advanced/theme/analytics.md]: Done in f000c15c2af1cbf5699e151c578176cfd3866487

**Comment by real34** [comment]
This PR is ready for a new review if you want. I only have to add theme screenshots for the 2.4 announcement.

Some highlights:
- 2.4 announcement at https://deploy-preview-145--heuristic-almeida-1a1f35.netlify.app/blog/2021/01/22/front-commerce-2.4/
- 2.3 announcement at https://deploy-preview-145--heuristic-almeida-1a1f35.netlify.app/blog/2021/01/13/front-commerce-2.3/
- up-to-date migration guide https://deploy-preview-145--heuristic-almeida-1a1f35.netlify.app/docs/appendices/migration-guides.html
- new "features" pages: https://deploy-preview-145--heuristic-almeida-1a1f35.netlify.app/docs/advanced/features/overview.html
- Log as customer documentation: https://deploy-preview-145--heuristic-almeida-1a1f35.netlify.app/docs/magento2/log-as-customer.html
- PM2 documentation: https://deploy-preview-145--heuristic-almeida-1a1f35.netlify.app/docs/advanced/production-ready/front-commerce-and-pm2.html
- … and minor other improvements elsewhere

File: source/_posts/front-commerce-2.4.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+## Meet Chocolatine, our new default theme
+
+We have been working on a new starter theme for future projects.
+Our goal was to improve our base theme so that new projects could start with a theme that was good enough to prototype and iterate.
+
+After some research, we've decided to leverage [the great UX and UI work from the ICEO team](https://www.behance.net/gallery/86012235/eCommerce-Free-UI-Kit) and use it as an starting point.
+It's name is "**Chocolatine**" *in reference to the correct way to pronounce "[Pain au chocolat](https://en.wikipedia.org/wiki/Pain_au_chocolat)" in French ;-)*
+
+This new theme provides a flexible foundation, ready to be adapted according to the merchant's brand. It allows to **deploy a new store in weeks instead of months**.
```
Comment (JulienPradet) [inline source/_posts/front-commerce-2.4.md:25]: Maybe mention the fact that some of our clients are already using it live?

File: source/_posts/front-commerce-2.4.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+We used the opportunity of this new theme to improve the overall accessibility:
+
+- better keyboard navigation,
+- direct access to the content,
+- more semantic markup usage,
+- mobile accessibility (touch events…)
+
+### Default eCommerce components in the core
+
+Over time, we detected that Front-Commerce projects often had to reimplemented some components not available in our default theme.
```
Comment (JulienPradet) [inline source/_posts/front-commerce-2.4.md:50]: Mega menu? Not really visible with the current sample data though.

File: source/docs/appendices/migration-guides.md [hunk 4]
```diff
@@ -7,6 +7,162 @@ This area will contain the Migration steps to follow for upgrading your store to
...
+## `2.3.x` -> `2.4.0`
+
+<blockquote class="important">
+  **We recommend that you first follow the [`2.2.x` -> `2.3.0` migration guide](#2-2-x-gt-2-3-0).**
+  This release was mostly focused on a new optional base theme, and there are not many upgrade instructions.
+</blockquote>
+
+### Cart item may not have product data attached
+
+We fixed some issues in case products added to cart went out of stock. Our default components were not resilient to Cart items not having an attached product anymore and led to errors.
```
Comment (JulienPradet) [inline source/docs/appendices/migration-guides.md:19]: The twe sentences deal about two different issues. Out of stock does not mean the product disappears. The disappearance issue was related to multiple website issues.

In the current form of this paragraph, it seems like the two issues are linked.

**Review by JulienPradet — COMMENTED** [review]
Thanks a lot for all this documentation!

**Pull request merged**